### DISCUSSION TITLE
Theming the component

### DISCUSSION THREAD

**Discussion by rafalkrupinski** [mskocik/svelty-picker#176] [General]
Hi
Can someone help with theming the component?

Looks like only the input and the buttons can be themed with css classes, while all the rest uses CSS variables.
Using CSS classes works ~~perfect~~ [edit: only for input, not for buttons] within my skeleton.dev (tailwind wrapper) project, but I'm having a hard time to set these variables to my theme variables.
Setting them to constants or to other variables within the same file works fine, but any other variables are ignored.

Example using accent colour to make it more visible:
```css
    --sdt-bg-main: rgb(var(--color-tertiary-500));
```

**Comment by rafalkrupinski** [comment]
example attempt to theme with CSS classes.
Only the inputClasses are honoured, buttons seem to lose styling.
```html
<SveltyPicker
        inputClasses="input w-[200px]"
        theme="text-secondary-50-900-token bg-primary-500-400-token border rounded shadow"
        todayBtnClasses="btn btn-primary"
        clearBtnClasses="btn btn-secondary"
/>
```

**Reply by mskocik** [reply]
Button classes are perfectly fine, see [example](https://svelte.dev/repl/a38bf505f9d749c7b349bfd68dd00be4?version=4.2.19), but you need to override them properly.

If you want to keep default styling you need to pass also default class/classes in `*BtnClasses` props.

**Reply by rafalkrupinski** [reply]
In the example above, I was trying to style the component with skeleton.dev classes. I didn't write them.

It was easier to override .std-component-wrap to use the theme variables

**Reply by mskocik** [reply]
Yes, that's what `std-component-wrap` is for. I thought your use case was slightly different.
